### PR TITLE
Add link to deprecation RFC in `no-array-prototype-extensions` rule doc

### PR DESCRIPTION
**Pull request by bmish**
TODO: update from [RFC PR](https://github.com/emberjs/rfcs/pull/848) to actual RFC link once merged.

CC: @smilland

### CODE REVIEW

**Comment by smilland** [comment]
RFC was merged! https://github.com/emberjs/rfcs/pull/848, feel free to merge it. Thank you @bmish!

**Comment by lin-ll** [comment]
Awesome! Let's update that link from the PR to the actual RFC link: https://rfcs.emberjs.com/id/0848-deprecate-array-prototype-extensions 
And then we'll merge it in! :) 
(I can do this later today if we're not in a rush) 

Thanks for your hard work on the RFC, @smilland !

File: docs/rules/no-array-prototype-extensions.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 By default, Ember extends certain native JavaScript objects with additional methods. This can lead to problems in some situations. One example is relying on these methods in an addon that is used inside an app that has the extensions disabled.
 
-The prototype extensions for the `Array` object will likely become deprecated in the future.
+The prototype extensions for the `Array` object were deprecated in [RFC #848](https://github.com/emberjs/rfcs/pull/848).
```
Comment (lin-ll) [inline docs/rules/no-array-prototype-extensions.md:5]:
```suggestion
The prototype extensions for the `Array` object were deprecated in [RFC #848](https://rfcs.emberjs.com/id/0848-deprecate-array-prototype-extensions).
```

File: docs/rules/no-array-prototype-extensions.md [hunk 2]
```diff
@@ -116,7 +116,7 @@ export default class SampleComponent extends Component {
 * [EmberArray](https://api.emberjs.com/ember/release/classes/EmberArray)
 * Ember [MutableArray](https://api.emberjs.com/ember/release/classes/MutableArray)
 * [Ember Prototype extensions documentation](https://guides.emberjs.com/release/configuring-ember/disabling-prototype-extensions/)
-* Ember Array prototype extensions deprecation RFC (TODO: add link when available)
+* [Ember Array prototype extensions deprecation RFC](https://github.com/emberjs/rfcs/pull/848)
```
Comment (lin-ll) [inline docs/rules/no-array-prototype-extensions.md:119]:
```suggestion
* [Ember Array prototype extensions deprecation RFC](https://rfcs.emberjs.com/id/0848-deprecate-array-prototype-extensions)
```

**Comment by bmish** [comment]
Thanks for finishing this up @lin-ll!

**Comment by smilland** [comment]
thank you @lin-ll !

**Pull request merged**